mongodb/eventstore.go

```diff
@@ -100,22 +159,31 @@ func (s *MongoDbEventStore) FromVersion(streamName goengine.StreamName, id strin
 }
 
 // CountEventsFor counts events for an id on the specified stream
-func (s *MongoDbEventStore) CountEventsFor(streamName goengine.StreamName, id string) (int, error) {
-	return s.db.C(string(streamName)).Find(bson.M{"aggregate_id": string(streamName)}).Count()
-}
+func (s *EventStore) CountEventsFor(streamName goengine.StreamName, id string) (int64, error) {
+	ctx, cancel := s.cs.CountEventsFor()
+	defer cancel()
 
-func (s *MongoDbEventStore) createIndexes(c *mgo.Collection) error {
-	index := mgo.Index{
-		Key:        []string{"aggregate_id", "version"},
-		Unique:     true,
-		DropDups:   true,
-		Background: true,
-	}
+	return s.mongoDB.Collection(string(streamName)).Count(ctx, bson.M{"aggregate_id": string(streamName)})
+}
 
-	return c.EnsureIndex(index)
+func (s *EventStore) createIndices(c *mongo.Collection) error {
```

Comment:
I think that here is createIndexes :)

Reply:
<img width="381" alt="image" src="https://user-images.githubusercontent.com/173697/52627864-a3e30300-2eb6-11e9-8a50-540aea1cb8ac.png">

Reply:
Sorry too much American :) didn't know that both are accepted